wolfcrypt/src/integer.c

```diff
@@ -4654,8 +4654,10 @@ int mp_prime_is_prime_ex (mp_int * a, int t, int *result, WC_RNG *rng)
         goto LBL_B;
     }
 
-    if (mp_cmp_d(&b, 2) != MP_GT || mp_cmp(&b, &c) != MP_LT)
+    if (mp_cmp_d(&b, 2) != MP_GT || mp_cmp(&b, &c) != MP_LT) {
+        ix--;
```

Comment:
Does tfm.c need the same fix, in mp_prime_is_prime_ex?

Reply:
No. The tfm.c code has a while loop.
I checked the implementation to be sure that the integer.c file was wrong.